client/rest/src/plugins/rosetta/nem/accountRoutes.js

```diff
@@ -0,0 +1,115 @@
+ * (at your option) any later version.
+ *
+ * Catapult is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Catapult.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { OperationParser } from './OperationParser.js';
+import { createLookupCurrencyFunction, getBlockchainDescriptor } from './rosettaUtils.js';
+import AccountBalanceRequest from '../openApi/model/AccountBalanceRequest.js';
+import AccountBalanceResponse from '../openApi/model/AccountBalanceResponse.js';
+import AccountCoinsRequest from '../openApi/model/AccountCoinsRequest.js';
+import AccountCoinsResponse from '../openApi/model/AccountCoinsResponse.js';
+import Amount from '../openApi/model/Amount.js';
+import BlockIdentifier from '../openApi/model/BlockIdentifier.js';
+import Coin from '../openApi/model/Coin.js';
+import CoinIdentifier from '../openApi/model/CoinIdentifier.js';
+import { RosettaErrorFactory, evaluateOperationsAndUpdateAmounts, rosettaPostRouteWithNetwork } from '../rosettaUtils.js';
+import { NetworkLocator } from 'symbol-sdk';
+import { Network } from 'symbol-sdk/nem';
+
+export default {
+	register: (server, db, services) => {
+		const blockchainDescriptor = getBlockchainDescriptor(services.config);
+		const network = NetworkLocator.findByName(Network.NETWORKS, blockchainDescriptor.network);
+		const lookupCurrency = createLookupCurrencyFunction(services.proxy);
+		const getChainHeight = () => services.proxy.fetch('chain/height', json => json.height);
+		const getBlockIdentifier = height => services.proxy.fetch('local/block/at', undefined, {
+			method: 'POST',
+			headers: { 'Content-Type': 'application/json' },
+			body: JSON.stringify({ height })
+		}).then(blockInfo => new BlockIdentifier(Number(blockInfo.block.height), blockInfo.meta.hash));
+		const parser = new OperationParser(network, {
+			includeFeeOperation: true,
+			lookupCurrency
+		});
+
+		const isCurrencyEqual = (lhs, rhs) => lhs.symbol === rhs.symbol && lhs.decimals === rhs.decimals;
+
+		const mapMosaicsToRosettaAmounts = mosaics => Promise.all(mosaics.map(async mosaic => {
+			const { currency } = await lookupCurrency(mosaic.mosaicId);
+			return new Amount(mosaic.quantity.toString(), currency);
+		}));
+
+		const getOperations = async unconfirmedTransactions => {
+			const mempoolTransactions = await Promise.all(unconfirmedTransactions.map(transaction =>
+				parser.parseTransactionAsRosettaTransaction(transaction.transaction, transaction.meta)));
+			return [].concat(...mempoolTransactions.map(transaction => transaction.operations));
+		};
+
+		const getAccountAmounts = async typedRequest => {
+			if (typedRequest.block_identifier)
+				throw RosettaErrorFactory.NOT_SUPPORTED_ERROR;
```

Comment:
This is fine if not supporting historical node

Reply:
would leave this until/unless we add historical node support